openedx-unsupported/configuration
# WIP: Make EDXAPP_ENV_EXTRA and EDXAPP_AUTH_EXTRA be included in LMS and CMS env files together with the specific settings

**Work in progress**. Do not merge yet. We'll ping when it goes through our internal review.

We found that when passing `EDXAPP_ENV_EXTRA` and `EDXAPP_LMS_ENV_EXTRA` together, settings from `EDXAPP_ENV_EXTRA` were discarded and only the ones from `EDXAPP_LMS_ENV_EXTRA` entered into `lms.env.json`.

We find it more useful if `EDXAPP_ENV_EXTRA` holds the common settings (LMS+CMS), and `EDXAPP_LMS_ENV_EXTRA` the LMS-specific ones, and then they are combined when creating `lms.env.json`.
This PR does that, and does the same for `EDXAPP_CMS_ENV_EXTRA`, and for the `….auth.json` files: `EDXAPP_AUTH_EXTRA`, `EDXAPP_LMS_AUTH_EXTRA`, `EDXAPP_CMS_AUTH_EXTRA`.


Configuration Pull Request
---

Make sure that the following steps are done before merging:


**Testing instructions**

TODO provide clearer steps and exact syntax.

Roughly:
1. Run the playbook with these variables: `EDXAPP_ENV_EXTRA` set to `{A: 1, B: 2}`, and `EDXAPP_LMS_ENV_EXTRA` to `{B: 22, C: 3}`.
2. The resulting `lms.env.json` should hold: `A: 1, B: 22, C: 3`. Before this patch, it would have been `B: 22, C: 3`.

**Reviewers**
